libs/blocks/z-pattern/z-pattern.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/*
+ * Z-Pattern - v5.1
+ */
+
+import { decorateBlockBg, getBlockSize } from '../../utils/decorate.js';
+import { decorateBlockAnalytics } from '../../utils/analytics.js';
+import media from '../media/media.js';
+
+export function decorateHeadline(header, size) {
+  const headingRow = header.parentElement;
+  headingRow.classList.add('heading-row');
+  headingRow.parentElement.classList.add('container');
+  const headerClass = (size === 'large') ? 'heading-XL' : 'heading-L';
+  header.classList.add(headerClass, 'headline');
+}
+
+function getReversedRowCount(rows) {
+  let count = 0;
+  rows.forEach((row) => {
+    const firstCol = row.querySelector(':scope > div > div:first-of-type');
+    const header = firstCol.querySelector('h1, h2, h3, h4, h5, h6');
+    // if first col has a header, its order is reversed
+    if (header) count += 1;
+  });
+  return count;
+}
+
+function getChildSingleRowCount(children) {
+  let length = 0;
+  for (let i = 0; i < children.length; i += 1) {
+    if (children[i].children.length === 1) {
+      length += 1;
+    }
+  }
+  return length;
```

Comment:
You could reduce here - in general we should try to avoid for loops.

```suggestion
  return [...children].reduce((length, child) => {
    if (child.children.length === 1) length += 1;
    return length;
  }, 0);
```